Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -274,7 +274,7 @@ static void WriteToHardware(u32 em_address, const u32 data, const u32 size)
 
   bool wi = false;
 
-  if (!never_translate && MSR.DR)
+  if (!never_translate && (IsOpcodeFlag(flag) ? MSR.IR : MSR.DR))
```

Comment:
IMO, I'd move this ternary out to it's own const local var instead of doing this inline in the conditional to make it a little nicer to read.

Reply:
That seems to have worked around the compiler error too.